### DISCUSSION TITLE
error defining specific class in state with factory function at the master branch

### DISCUSSION THREAD

**Discussion by gkyildirim** [ngrx/platform#4622] [Q&A] [answered]
### Which @ngrx/* package(s) are the source of the bug?

signals

### Minimal reproduction of the bug/regression with instructions

At the commit e2f35c877146635e44a2dd99d642980277b36620 of signal store if a formgroup is defined as a state by factory function  we have the following error.

```js
BehaviorSubject.js:34 Uncaught TypeError: Cannot assign to read only property '_value' of object '#<BehaviorSubject2>'

core.mjs:5907 Uncaught (in promise) TypeError: Cannot assign to read only property 'taskId' of object '#<_PendingTasksInternal>'

```

It is reproducable as
```ts
 withState(() => {
    return { formTest: new FormGroup({}) };
  }),
  ```
  
However this works
```ts
withState({ formTest: new FormGroup({}) }),
```  

### Expected behavior

With 18.1.1 this is working. 

### Versions of NgRx, Angular, Node, affected browser(s) and operating system(s)

Ngrx: e2f35c877146635e44a2dd99d642980277b36620
Angular: 19.0.3
Node: v22.12.0
Chrome in macOS.

### Other information

I guess upcoming withProps() would be a good practice to use forms in signal store but I post this in case it helps since there might be potential cases other than formgroups.

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

**Comment by markostanimirovic** [comment, marked as answer]
The `FormGroup` instance or any other object with methods, observables, etc. should not be part of the SignalStore's state. The state should be a plain object instead.

**Comment by gkyildirim** [comment]
Thanks for the explanation. This was a trial to migrate existing project. I guess withProps will make a fit here?

**Reply by markostanimirovic** [reply]
Yes, that's right 👍
